IPython/html/static/tree/js/notebooklist.js

```diff
@@ -382,78 +386,26 @@ define([
     };
 
     /**
-     * Select all of the items in the tree.
+     * Select all items in the tree of specified type.
+     * selection_type : string among "select-all, "select-folders", "select-notebooks", "select-running-notebooks", "select-files"
+     * state : boolean, true to select and false to deselect
      */
-    NotebookList.prototype.select_all = function() {
-        $('.list_item input[type=checkbox]').each(function(index, item) {
-            $(item).prop('checked', true);
-        });
-        this._selection_changed();
-    };
-
-    /**
-     * Select all of the notebooks in the tree.
-     */
-    NotebookList.prototype.select_notebooks = function() {
-        this.deselect_all();
-        $('.list_item').each(function(index, item) {
-            if ($(item).data('type') === 'notebook') {
-                $(item).find('input[type=checkbox]').prop('checked', true);
-            }
-        });
-        this._selection_changed();
-    };
-
-    /**
-     * Select all of the running notebooks in the tree.
-     */
-    NotebookList.prototype.select_running_notebooks = function() {
-        this.deselect_all();
+    NotebookList.prototype.select = function(selection_type,state) {
         var that = this;
         $('.list_item').each(function(index, item) {
-            if ($(item).data('type') === 'notebook' && that.sessions[$(item).data('path')] !== undefined) {
-                $(item).find('input[type=checkbox]').prop('checked', true);
-            }
-        });
-        this._selection_changed();
-    };
-
-    /**
-     * Select all of the files in the tree.
-     */
-    NotebookList.prototype.select_files = function() {
-        this.deselect_all();
-        $('.list_item').each(function(index, item) {
-            if ($(item).data('type') === 'file') {
-                $(item).find('input[type=checkbox]').prop('checked', true);
+            // For each item, determine if the state should be set, depending on the selection_type that triggered select
+            var set_state = (selection_type === "select-all");
+            set_state = set_state || (selection_type === "select-folders" && $(item).data('type') === 'directory');
+            set_state = set_state || (selection_type === "select-notebooks" && $(item).data('type') === 'notebook');
+            set_state = set_state || (selection_type === "select-running-notebooks" && $(item).data('type') === 'notebook' && that.sessions[$(item).data('path')] !== undefined);
+            set_state = set_state || (selection_type === "select-files" && $(item).data('type') === 'file');
```

Comment:
I have a minor style request here.  These lines are a little long, partially because you keep fetching `$(item).data('type')`.  Could you put that in a variable, like `type` or `item_type`?